src/utils.ts

```diff
@@ -327,5 +327,27 @@ export class Utils {
       }
     }
   }
+
+  /** @internal */

```

Comment:
put a real comment on your code. seems reasonable given we do something similar (though quite different code. I will have to diff into the scroll as I didn't write that)